File: PartyProfit.cpp
#include <iostream>

std::pair<int, int> calculateCoinsPerCompanion(int numberOfCompanions, int days) {
    int coins = 0;
    for (int currentDay = 1; currentDay <= days; ++currentDay) {
        if (currentDay % 10 == 0) {
            numberOfCompanions -= 2;
        }
        if (currentDay % 15 == 0) {
            numberOfCompanions += 5;
        }
        if (currentDay % 3 == 0) {
            coins -= numberOfCompanions * 3;
        }
        if (currentDay % 5 == 0) {
            coins += numberOfCompanions * 20;
            if (currentDay % 3 == 0) {
                coins -= numberOfCompanions * 2;
            }
        }
        coins += 50;
        coins -= numberOfCompanions * 2;
    }
    int coinsPerCompanion = coins / numberOfCompanions;
    return std::make_pair(coinsPerCompanion, numberOfCompanions);
}

int main() {
    int numberOfCompanions, days;
    std::cout << "Enter the number of companions: ";
    std::cin >> numberOfCompanions;
    std::cout << "Enter the number of days: ";
    std::cin >> days;

    std::pair<int, int> result = calculateCoinsPerCompanion(numberOfCompanions, days);
    int coinsPerCompanion = result.first;
    numberOfCompanions = result.second;

    std::cout << numberOfCompanions << " companions received " << coinsPerCompanion << " coins each." << std::endl;
    return 0;
}

File: Party_Profit.py
def calculate_coins_per_companion(number_of_companions, days):
  """Calculates the number of coins each companion receives, based on the given number of companions and days.

  Args:
    number_of_companions(int): The number of companions.
    days(int): The number of days.

  Returns:
    The number of coins each companion receives.
  """

  coins = 0
  for current_day in range(1, days + 1):
    if current_day % 10 == 0:
      number_of_companions -= 2
    if current_day % 15 == 0:
      number_of_companions += 5
    if current_day % 3 == 0:
      coins -= number_of_companions * 3
    if current_day % 5 == 0:
      coins += number_of_companions * 20
      if current_day % 3 == 0:
        coins -= number_of_companions * 2
    coins += 50
    coins -= number_of_companions * 2

  coins_per_companion = coins // number_of_companions
  return coins_per_companion, number_of_companions


# Get the number of companions and days from the user.
number_of_companions = int(input("Enter the number of companions: "))
days = int(input("Enter the number of days: "))

# Calculate the number of coins each companion receives.
coins_per_companion, number_of_companions = calculate_coins_per_companion(number_of_companions, days)

# Print the results.
print(f"{number_of_companions} companions received {coins_per_companion} coins each.")
